homeassistant/components/google_assistant/__init__.py

```diff
@@ -73,7 +73,7 @@ def _check_report_state(data):
             ): cv.ensure_list,
             vol.Optional(CONF_ENTITY_CONFIG): {cv.entity_id: ENTITY_SCHEMA},
             # str on purpose, makes sure it is configured correctly.
-            vol.Optional(CONF_SECURE_DEVICES_PIN): str,
+            vol.Optional(CONF_SECURE_DEVICES_PIN): vol.Any(str, [str]),
```

Comment:
Should we ensure a list here instead? That way, we don't have to check for `isinstance(pin, list)` in the code processing this value

Reply:
Wouldn't that break backwards compatibility with existing users?

Reply:
No? It would just ensure the output if this configuration parameter is a list, one can still use a single value string value. Search our codebase for `cv.ensure_list` for a bunch of example usages.

Reply:
ensure list wraps it in a list if it's not one. It doesn't enforce lists.

Reply:
Ah thanks for explaining, wasn't aware of a `cv.ensure_list`, thought you meant using `[str]`. Will make those changes.

Reply:
Fixed in Fixed in https://github.com/home-assistant/core/pull/64429/commits/3859310c4ee598780fc55b0aff088c16ff264735